src/components/Modals/QrCode/Form.tsx

```diff
@@ -147,42 +146,41 @@ export const Form: React.FC<QrCodeFormProps> = ({ accountId }) => {
   )
 
   useEffect(() => {
-    history.push(SendRoutes.Scan)
+    navigate(SendRoutes.Scan)
   }, [history])
 
   if (walletConnectDappUrl)
     return <ConnectModal initialUri={walletConnectDappUrl} isOpen={isOpen} onClose={handleClose} />
 
+  const selectAssetRouterElement = useMemo(
+    () => <SelectAssetRouter onBack={handleBack} onClick={handleAssetSelect} />,
+    [handleBack, handleAssetSelect],
+  )
+
+  const addressElement = useMemo(() => <Address />, [])
+  const detailsElement = useMemo(() => <Details />, [])
+  const qrCodeScannerElement = useMemo(
+    () => (
+      <QrCodeScanner onSuccess={handleQrSuccess} onBack={handleClose} addressError={addressError} />
+    ),
+    [handleClose, handleQrSuccess, addressError],
+  )
+  const confirmElement = useMemo(() => <Confirm />, [])
+  const statusElement = useMemo(() => <Status />, [])
+
   return (
     <FormProvider {...methods}>
-      {/* eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions */}
       <form onSubmit={methods.handleSubmit(handleSubmit)} onKeyDown={checkKeyDown}>
         <AnimatePresence mode='wait' initial={false}>
-          <Switch location={location} key={location.key}>
-            <Route path={SendRoutes.Select}>
-              <SelectAssetRouter onBack={handleBack} onClick={handleAssetSelect} />
-            </Route>
-            <Route path={SendRoutes.Address}>
-              <Address />
-            </Route>
-            <Route path={SendRoutes.Details}>
-              <Details />
-            </Route>
-            <Route path={SendRoutes.Scan}>
-              <QrCodeScanner
-                onSuccess={handleQrSuccess}
-                onBack={handleClose}
-                addressError={addressError}
-              />
-            </Route>
-            <Route path={SendRoutes.Confirm}>
-              <Confirm />
-            </Route>
-            <Route path={SendRoutes.Status}>
-              <Status />
-            </Route>
-            <Route path='/' exact render={scanRedirect} />
-          </Switch>
+          <Routes>
+            <Route path={`${SendRoutes.Select}/*`} element={selectAssetRouterElement} />
+            <Route path={SendRoutes.Address} element={addressElement} />
+            <Route path={SendRoutes.Details} element={detailsElement} />
+            <Route path={SendRoutes.Scan} element={qrCodeScannerElement} />
+            <Route path={SendRoutes.Confirm} element={confirmElement} />
+            <Route path={SendRoutes.Status} element={statusElement} />
+            <Route path='/' element={scanRedirect} />
+          </Routes>
```

Comment:
Yeah, somehow this works, that's the memory routers unpredictability I did mention above - works in very rare cases, but usually doesn't.